weave/wandb_interface/wandb_lite_run.py

```diff
@@ -51,6 +51,7 @@ def __init__(
         group: typing.Optional[str] = None,
         _hide_in_wb: bool = False,
         _use_async_file_stream: bool = False,
+        _optimize_for_artifact_commit: bool = False,
```

Comment:
`optimize_for_artifact_commit` as a variable name implies that the lite run knows what is needed to optimize for artifact committing. Feels like a bleed of responsibilities. I think this could be modified with 2 changes:
1. it might be called: `_create_project_if_needed` or something like that and default to `true`. Which will do exactly what the variable says
2. The step initialization:
```
            self._step = 0
            if not self._optimize_for_artifact_commit:
                self._step = self._run.lastHistoryStep + 1
```
can be refactored to lazily initialize when needed. This will remove the dependency on this constructor param.

Reply:
I don't think the lazy init on history step works here. When the Filepusher is initialized, it also initializes Filestream which uses the history step with the following code:

```python
self._stream.set_file_policy(
    "wandb-history.jsonl",
    file_stream.JsonlFilePolicy(start_chunk_id=**self._step**),
)
```

We need to disambiguate which context LiteRun is used under. So I think the boolean flag is the way to go. 

I'll do two boolean flags which both default to true:
- `create_project`
- `get_last_history_step`